cmake_modules/CPackUtilities.cmake

```diff
@@ -0,0 +1,117 @@
+# create win-exe, dmg, and deb
+#-------------------------------
+# Structure:
+# ├── LICENSE
+# ├── README.md
+# ├── JuPedSim.pdf
+# ├── bin
+# │   ├── jpscore
+# │   └── jpsreport
+# ├── jpscore_samples
+# └── jpsreport_samples
+#-------------------------------
+function (cpack_write_deb_config)
+  message(STATUS "Package generation - LINUX")
+  list(APPEND CPACK_GENERATOR "DEB")
+  set (CPACK_GENERATOR  ${CPACK_GENERATOR} PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_DEPENDS "libboost-dev (>=1.65), zlib1g-dev" PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "http://jupedsim.org" PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_SUGGESTS, "jpsvis, jpseditor" PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_PRIORITY "optional" PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_SECTION "science" PARENT_SCOPE)
+  set(CPACK_DEBIAN_ARCHITECTURE ${CMAKE_SYSTEM_PROCESSOR} PARENT_SCOPE)
+  set( CPACK_DEBIAN_PACKAGE_MAINTAINER "David Haensel" PARENT_SCOPE)
+  set(CPACK_DEBIAN_PACKAGE_DESCRIPTION "JuPedSim: framework for simulation and analysis of pedestrian dynamics" PARENT_SCOPE)
+endfunction()
+
+function (cpack_write_osx_config)
+  message(STATUS "Package generation - MacOS")
+  list(APPEND CPACK_GENERATOR "DragNDrop")
+  set (CPACK_GENERATOR  ${CPACK_GENERATOR} PARENT_SCOPE)
+  Set(CPACK_DMG_BACKGROUND_IMAGE "${CMAKE_SOURCE_DIR}/jpscore/forms/background.png" PARENT_SCOPE)
+   set(CPACK_DMG_DS_STORE_SETUP_SCRIPT
+     "${CMAKE_SOURCE_DIR}/jpscore/forms/DS_Store.scpt" PARENT_SCOPE)
+  set(CPACK_DMG_DISABLE_APPLICATIONS_SYMLINK ON  PARENT_SCOPE)
+  set(CPACK_PACKAGE_ICON "${CMAKE_SOURCE_DIR}/jpscore/forms/JPScore.icns" PARENT_SCOPE)
+  set(CPACK_DMG_VOLUME_NAME "${PROJECT_NAME}" PARENT_SCOPE)
+  set(CPACK_SYSTEM_NAME "OSX" PARENT_SCOPE)
+endfunction()
+
+function (cpack_write_windows_config)
+  # executable needs two dlls: zlib1.dll and fmt.dll
+  # these are copied along with the system dlls
+  # NOTE: Assume installation with vcpkg
+  message(STATUS "Package generation - Windows")
+  list(APPEND CPACK_GENERATOR "NSIS")
+  set (CPACK_GENERATOR  ${CPACK_GENERATOR} PARENT_SCOPE)
+  find_program(NSIS_PATH nsis PATH_SUFFIXES nsis)
+  if(NSIS_PATH)
+    message(STATUS "   + NSIS                                 YES ")
+  else(NSIS_PATH)
+    message(STATUS "   + NSIS                                 NO ")
+  endif(NSIS_PATH)
+
+  #set(CMAKE_INSTALL_MFC_LIBRARIES ON)
+  set(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP TRUE)
+  set(VCPKG_DIR "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/bin/")
+  set(DIRS ${VCPKG_DIR})
+  #file(GLOB DLL_FILES "${VCPKG_DIR}/*.dll")
```

Comment:
commented out code :) this can be removed

Reply:
done